ring/partition_ring.go

```diff
@@ -340,3 +352,60 @@ func (r *PartitionRing) String() string {
 
 	return fmt.Sprintf("PartitionRing{ownersCount: %d, partitionsCount: %d, partitions: {%s}}", len(r.desc.Owners), len(r.desc.Partitions), buf.String())
 }
+
+// ActivePartitionBatchRing wraps PartitionRing and implements DoBatchRing to lookup ACTIVE partitions.
+type ActivePartitionBatchRing struct {
+	ring *PartitionRing
+}
+
+func NewActivePartitionBatchRing(ring *PartitionRing) *ActivePartitionBatchRing {
```

Comment:
shouldn't the `Operation` parameter to `Get` determine which instances (partitions in our case) get included in the replication set? Having a ring implementation for a specific partition state seems different from how the classic ring and DoBatch work.

We do have PartitionState and not InstanceStates, but maybe we can implement a mapping for the purposes of an Operation.

Reply:
another option is to extend `Operation` to also support these methods, which are similar to the existing methods for `InstanceState`


```go
func (op Operation) IsPartitionInStateHealthy(s PartitionState) bool {}
func (op Operation) ShouldExtendReplicaSetOnPartitionState(s PartitionState) bool {}
```

Reply:
> shouldn't the Operation parameter to Get determine which instances (partitions in our case) get included in the replication set?

Yes. I think we can solve the problem with generics (I've a local copy based on generics that looks working). Let me merge this PR as is and then open a follow up PR, so that I'm not blocked on other PRs based on this.

Reply:
i'm not even sure we need generics - why not just reuse `Operation`? it's only a `uint32`